In [3]:
import seaborn as sns
import numpy as np
import pandas as pd
import scipy as sp
import plotly.express as px
import plotly.graph_objects as go
import plotly.figure_factory as ff
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [4]:
df = pd.read_csv("/content/drive/MyDrive/DA_Dataset/ste11.csv")


In [5]:
df.shape

(65, 8)

In [8]:
df.head()

Country_code    Country  Total Library Size  No. of TV Shows  No. of Movies  \
0           ar  Argentina                4760             3154           1606   
1           au  Australia                6114             4050           2064   
2           at    Austria                5640             3779           1861   
3           be    Belgium                4990             3374           1616   
4           bo    Bolivia                4991             3155           1836   

   Cost Per Month - Basic ($)  Cost Per Month - Standard ($)  \
0                        3.74                           6.30   
1                        7.84                          12.12   
2                        9.03                          14.67   
3                       10.16                          15.24   
4                        7.99                          10.99   

   Cost Per Month - Premium ($)  
0                          9.26  
1                         16.39  
2                         20.32  
3                         20.32  
4                         13.99

In [9]:
df.tail()

Country_code         Country  Total Library Size  No. of TV Shows  \
60           ua         Ukraine                5336             3261   
61           gb  United Kingdom                6643             4551   
62           us   United States                5818             3826   
63           uy         Uruguay                4989             3154   
64           ve       Venezuela                4797             3154   

    No. of Movies  Cost Per Month - Basic ($)  Cost Per Month - Standard ($)  \
60           2075                        5.64                           8.46   
61           2092                        7.91                          13.20   
62           1992                        8.99                          13.99   
63           1835                        8.99                          12.99   
64           1643                        7.99                          10.99   

    Cost Per Month - Premium ($)  
60                         11.29  
61                         18.48  
62                         17.99  
63                         15.99  
64                         13.99

In [10]:
df = df.drop(columns=['Country_code'], axis=1)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65 entries, 0 to 64
Data columns (total 7 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   Country                        65 non-null     object 
 1   Total Library Size             65 non-null     int64  
 2   No. of TV Shows                65 non-null     int64  
 3   No. of Movies                  65 non-null     int64  
 4   Cost Per Month - Basic ($)     65 non-null     float64
 5   Cost Per Month - Standard ($)  65 non-null     float64
 6   Cost Per Month - Premium ($)   65 non-null     float64
dtypes: float64(3), int64(3), object(1)
memory usage: 3.7+ KB


Data has no null values

In [11]:
df.describe()

Total Library Size  No. of TV Shows  No. of Movies  \
count           65.000000        65.000000      65.000000   
mean          5314.415385      3518.953846    1795.461538   
std            980.322633       723.010556     327.279748   
min           2274.000000      1675.000000     373.000000   
25%           4948.000000      3154.000000    1628.000000   
50%           5195.000000      3512.000000    1841.000000   
75%           5952.000000      3832.000000    1980.000000   
max           7325.000000      5234.000000    2387.000000   

       Cost Per Month - Basic ($)  Cost Per Month - Standard ($)  \
count                   65.000000                      65.000000   
mean                     8.368462                      11.990000   
std                      1.937819                       2.863979   
min                      1.970000                       3.000000   
25%                      7.990000                      10.710000   
50%                      8.990000                      11.490000   
75%                      9.030000                      13.540000   
max                     12.880000                      20.460000   

       Cost Per Month - Premium ($)  
count                     65.000000  
mean                      15.612923  
std                        4.040672  
min                        4.020000  
25%                       13.540000  
50%                       14.450000  
75%                       18.060000  
max                       26.960000

From above command, pandas readily provided the important statistical data related to my data. Here, I found that among all the 65 countries which are beneficiaries of Netflix's services, 2274 is the lowest library size that Netflix provides to any country whereas 7325 is the highest library size provided. Thus, making the range 5051. This also provided me the value of mean and median(50%) of each column.

In [12]:
# Outlier detection in Total Library Size, No. of TV Shows, No. of Movies columns 
fig = px.box(df, y=df.columns[1:4])
fig.show()

In [13]:
# Outlier detection in Cost Per Month - Basic ($), Cost Per Month - Standard ($), Cost Per Month - Premium ($) columns 
fig = px.box(df, y=df.columns[4:])
fig.show()

Above boxplots, provided the information regarding the distribution of data. From first plot we can observer no. of TV shows available on NETFLIX is higher than movies. From second plot, we can observer that price distribution rises as we move from basic to premium plan through standard. Also, we can conclude that, 50% of the Basic plan prices lie in the range of 90.3 to 7.97 USD. These, boxplot shows the outliers that are present in the data (points lying outside the boxplot whiskers).

In [14]:
hist_data = [df['Total Library Size']]

group_labels = ['Total Library Size']

fig = ff.create_distplot(hist_data, group_labels, show_hist=False, curve_type='normal', bin_size=0.5, show_rug=False)
fig.show()

In [15]:
hist_data = [df['No. of TV Shows']]

group_labels = ['No. of TV Shows']

fig = ff.create_distplot(hist_data, group_labels, show_hist=False, curve_type='normal', bin_size=0.5, show_rug=False)
fig.show()

In [16]:
hist_data = [df['Cost Per Month - Basic ($)']]

group_labels = ['Cost Per Month - Basic ($)']

fig = ff.create_distplot(hist_data, group_labels, show_hist=False, curve_type='normal', bin_size=0.5, show_rug=False)
fig.show()

Normal distribution shows the most accurate distribution of data around its mean. On comparing the value at tip and the value given by pandas describe function, I realized that they are same. Which verifies that mean is at peak in normal distribution.

In [17]:
df.corr()

Total Library Size  No. of TV Shows  \
Total Library Size                       1.000000         0.971146   
No. of TV Shows                          0.971146         1.000000   
No. of Movies                            0.849957         0.699786   
Cost Per Month - Basic ($)              -0.062877        -0.003814   
Cost Per Month - Standard ($)           -0.055200         0.002285   
Cost Per Month - Premium ($)            -0.071521        -0.011100   

                               No. of Movies  Cost Per Month - Basic ($)  \
Total Library Size                  0.849957                   -0.062877   
No. of TV Shows                     0.699786                   -0.003814   
No. of Movies                       1.000000                   -0.179913   
Cost Per Month - Basic ($)         -0.179913                    1.000000   
Cost Per Month - Standard ($)      -0.170391                    0.898535   
Cost Per Month - Premium ($)       -0.189711                    0.820914   

                               Cost Per Month - Standard ($)  \
Total Library Size                                 -0.055200   
No. of TV Shows                                     0.002285   
No. of Movies                                      -0.170391   
Cost Per Month - Basic ($)                          0.898535   
Cost Per Month - Standard ($)                       1.000000   
Cost Per Month - Premium ($)                        0.978768   

                               Cost Per Month - Premium ($)  
Total Library Size                                -0.071521  
No. of TV Shows                                   -0.011100  
No. of Movies                                     -0.189711  
Cost Per Month - Basic ($)                         0.820914  
Cost Per Month - Standard ($)                      0.978768  
Cost Per Month - Premium ($)                       1.000000

Above is the correlation matrix which gives us the dependence of one attribute on another basically their relation. If correlation is near to +1 then it is understood that varying the value of one attribute varies another attribute in the same direction where as if it is near to -1 then, it is understood that varying one value will vary another value in oposite direction. E.g. Total Library Size = No. of TV Shows + No. of Movies, hence we found the correlation value of 0.97 (strong +ve correlation).

In [19]:
print('\n\nTotal Library Size\n', df['Total Library Size'].mode())
print('\n\nNo. of TV Shows\n', df['No. of TV Shows'].mode())
print('\n\nNo. of Movies\n', df['No. of Movies'].mode())
print('\n\nCost Per Month - Basic ($)\n', df['Cost Per Month - Basic ($)'].mode())
print('\n\nCost Per Month - Standard ($)\n', df['Cost Per Month - Standard ($)'].mode())
print('\n\nCost Per Month - Premium ($)\n', df['Cost Per Month - Premium ($)'].mode())



Total Library Size
 0    4797
1    4989
2    4991
dtype: int64


No. of TV Shows
 0    3154
dtype: int64


No. of Movies
 0    1835
1    1971
dtype: int64


Cost Per Month - Basic ($)
 0    9.03
dtype: float64


Cost Per Month - Standard ($)
 0    11.29
1    14.67
dtype: float64


Cost Per Month - Premium ($)
 0    20.32
dtype: float64


Above is the mode of each attribute, showing values with highest frequencies in their respective columns.

In [20]:
df.describe()


Total Library Size  No. of TV Shows  No. of Movies  \
count           65.000000        65.000000      65.000000   
mean          5314.415385      3518.953846    1795.461538   
std            980.322633       723.010556     327.279748   
min           2274.000000      1675.000000     373.000000   
25%           4948.000000      3154.000000    1628.000000   
50%           5195.000000      3512.000000    1841.000000   
75%           5952.000000      3832.000000    1980.000000   
max           7325.000000      5234.000000    2387.000000   

       Cost Per Month - Basic ($)  Cost Per Month - Standard ($)  \
count                   65.000000                      65.000000   
mean                     8.368462                      11.990000   
std                      1.937819                       2.863979   
min                      1.970000                       3.000000   
25%                      7.990000                      10.710000   
50%                      8.990000                      11.490000   
75%                      9.030000                      13.540000   
max                     12.880000                      20.460000   

       Cost Per Month - Premium ($)  
count                     65.000000  
mean                      15.612923  
std                        4.040672  
min                        4.020000  
25%                       13.540000  
50%                       14.450000  
75%                       18.060000  
max                       26.960000

In [21]:
# alpha = 0.05
# H0: u = 8
# Ha: u > 8
sample = df['Cost Per Month - Basic ($)'].sample(8)
print("Sample:\n",sample)
print("\n\ns:", 1.937819)
print("x-bar:", sample.mean())
print("n: 8")
SE = 1.937819/np.sqrt(8)
print("SE: ", SE)
t_stat, p_value = sp.stats.ttest_1samp(a=sample, popmean=8)
print("p-value:", p_value, "\nt-stat", t_stat)
# Zstat = (sample.mean() - 8)/ SE
# print("Zstat: ", Zstat)
if t_stat > 1.895:
  print("H0 rejected")
else:
  print("H0 accepted")



if p_value >0.1:
  print("Non significant evidence against H0")
elif p_value > 0.05 and p_value <= 0.1:
  print("Marginally significant evidence against H0") 
elif p_value > 0.01 and p_value <= 0.05:
  print("Significant evidence against H0") 
else:
  print("Highly significant evidence against H0") 

Sample:
 17    10.16
47     9.03
6      9.03
37     9.03
22     7.99
11     9.03
38     9.03
63     8.99
Name: Cost Per Month - Basic ($), dtype: float64


s: 1.937819
x-bar: 9.03625
n: 8
SE:  0.685122477806067
p-value: 0.0014830809845042749 
t-stat 5.04803211189779
H0 rejected
Highly significant evidence against H0


Here, I am performing the hypothesis test using z-score and p-value
If Zstat > Zalpha, H0 gets rejected else accepted. Also, if p-value is less than alpha then H0 is rejected else accepted.


Solution to example given in the document

Given data:

n = 10

x-bar = 27

variance = 20

H0: u0 = 30

Ha: ua != 30

alpha = 0.05

Solution:

As standar deviation is given, I'll perform z-test to test the hypothesis

SE = sqrt(variance)/sqrt(n) = sqrt(20)/sqrt(10) = 1.414

zstat = (x-bar - u0)/SE = (-3)/1.414 = -2.122

Therefore, p-value = 0.033838
as, p-value is between 0.01 and 0.05, we have significant evidence against H0.
As, p-value: 0.033838 is lesseer than alpha: 0.05, we will reject the null hypothesis.

To check whether u < 30

H0: u0 = 30

Ha: ua < 30

alpha = 0.05

Solution:

SE = sqrt(variance)/sqrt(n) = sqrt(20)/sqrt(10) = 1.414

zstat = (x-bar - u0)/SE = (-3)/1.414 = -2.122

As this is a left tailed test,

We need to check whether Zstat < -Zalpha

If true, we reject H0 else we retain it.

Here, Zalpha = -1.645

As, Zstat: -2.122 < -Zalpha: -1.645

Therefore, we reject null hypothesis and hold u < 30 as True.

**Conclusion**

Descriptive Statistical Analysis provides insights of data which can be easily understood by analyst. It provides the information such that a distinct pattern may arise out of that. It gives conclusion of distribution of our data, helps in detection of outliers, mean, median and mode. It also enables us to identify the relation between different attributes of our data.

In this example, standard deviation was not given also the sample size was 8 << 30. Hence, I chose to go for t-test to test my hypothesis. If t-stat > t-alpha, mean lies in rejection region (right-tailed test). Hence H0 is rejected. I also observed that, lesser the p-value more are the chances of rejecting the null hypothesis. Using this test, I can prove my null hypothesis to be true or false.